### DISCUSSION TITLE
Coexistence of original APP and API client

### DISCUSSION THREAD

**Discussion by kaischmitz** [SAIC-iSmart-API/documentation#5] [General]
I understand by SAIC's philosophy there can be only one control instance bound to one car. So far so bad.
I noticed that while using mqtt_gateway.py I get thrown out of the app on my phone, too.
Is there any way to have both running in parallel, e.g. by making the server think the request are coming from the same source?

**Comment by tisoft** [comment]
You would need to sniff the traffic between the App and the SAIC servers while you are logged into the App. Copy out the token und use that in the MQTT gateway. Currently this would be possible, because the token doesn't seem to expire. But as soon as SAIC changes this behavior, you would need to do the token sniffing again.
For the Java MQTT gateway I plan to detect, when it gets logged out, when the App is logging in and pause for a few minutes, so that it's possible to use the App and login after a few minutes. Also I plan to support all features the App provides, so that it should not be necessary to start the App for the most time.

**Reply by kaischmitz** [reply]
How do I extract the token from the captured messages? The reverse-engineering README.md seems to lack some details about it.
I would expect the token to be in the request part of the message(s) going to https://tap-eu.soimt.com/TAP.Web/ota.mpv21, is that assumption correct?
I tried to modify the old decodeSampleResponse.py to use more recent ASN.1 schema files, but the results I'm getting still look kind of arbitrary.
